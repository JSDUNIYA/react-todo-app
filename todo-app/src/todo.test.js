import {render, screen, cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer';
import Todo from './todo'

let data = {
    id:1,
    name: "My first task",
    done: true,
};

let data2 = {
    id:1,
    name: "My first task",
    done: false,
};


afterEach(()=>{
    cleanup();
})
test('Render todo', () => {
    render(<Todo todoData={data}/>)
    const todoElement = screen.getByTestId('data-1');

    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('My first task');
    expect(todoElement).not.toContainHTML('<strike>')
})

test('match Snapshot', () =>{
    const tree = renderer.create(<Todo todoData={data}/>).toJSON();
    expect(tree).toMatchSnapshot();

})