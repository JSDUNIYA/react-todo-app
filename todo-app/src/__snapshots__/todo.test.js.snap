// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match Snapshot 1`] = `
<div
  data-testid="data-1"
>
  <strike>
    <h1>
      My first task
    </h1>
  </strike>
</div>
`;
